packages/node_modules/@webex/plugin-meetings/src/reachability/index.js

```diff
@@ -182,6 +192,76 @@ export default class Reachability extends StatelessWebexPlugin {
     });
   }
 
+
+  /**
+   * An event handler on an RTCPeerConnection when the state of the ICE
+   * candidate gathering process changes. Used to measure connection
+   * speed.
+   * @private
+   * @param {RTCPeerConnection} peerConnection
+   * @returns {Promise}
+   */
+  iceGatheringState = (peerConnection) =>
+    new Promise((resolve) => {
+      const waitTime = 1e4;
+      const startKey = `start:${peerConnection.key}`;
+      const endKey = `end:${peerConnection.key}`;
+      const getElapsedTime = () => {
+        const percision = 3;
+        const elapsed = window.performance.measure(startKey, endKey).duration;
+
+        return Number.isNaN(percision) ? elapsed : parseFloat(elapsed.toFixed(percision));
+      };
+
+      let timeout = setTimeout(() => {
+        LoggerProxy.logger.error(`Reachability:index#iceGatheringState --> No ice candidate for ${peerConnection.clusterId}.`);
+        resolve();
+      }, waitTime);
+
+      // eslint-disable-next-line no-param-reassign
+      peerConnection.addEventListener('icegatheringstatechange', () => {
```

Comment:
In short here we are using either the icegatheringState change or onIcecandidate to calculate the time